### PR TITLE
Add Infra (K8s Config, Dockerfile, Fix Nits, and Docker Compose)

### PR DESCRIPTION
**Pull request by joyliu-q**
- Add new Dockerfile
- Add to Docker Compose (corrected some nits)
- Add K8s config
- Fix nits in naming
- Add GitHub Actions
- Make changes for Uvicorn to run in Docker

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+FROM python:3.10-buster
```
Comment (judtinzhang) [inline Dockerfile:1]: could we use `python:3.11-slim-buster` instead?

File: .github/build-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+
+# ========================================
+# Note:  If you make changes to this CI/CD, please include someone from DevOps in the list of reviewers for the PR.
+# ========================================
+name: Build and Deploy PennCourses
```
Comment (judtinzhang) [inline .github/build-deploy.yaml:5]: nit: `PennCourses`

File: .github/build-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+jobs:
+  backend-check:
+    name: "Backend Check"
+    uses: pennlabs/shared-actions/.github/workflows/django.yaml@8785a7d7b9158d8d5705a0202f5695db2c0beb97
+    with:
+      projectName: LabsAnalytics
+      path: .
+      flake: true
+      black: true
+      pythonVersion: 3.10-buster
```
Comment (judtinzhang) [inline .github/build-deploy.yaml:18]: nit: can we use 3.11

File: Dockerfile (hunk 1 above)
Comment (judtinzhang) [reply Dockerfile:1]: or any other 3.11 image you like

File: k8s/main.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        image: backendImage,
+        secret,
+        replicas: 1,
+        // TODO: check fastapi env variables
+        env: [],
+      },
+      ingress: {
+        rules: [{
+          host: domain,
+          paths: ["/analytics"], // TODO: add paths here
```
Comment (judtinzhang) [inline k8s/main.ts:40]: this is the only route, so we are good here!

File: k8s/main.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      persistData: true,
+    });
+
+    new Application(this, 'backend', {
+      deployment: {
+        image: backendImage,
+        secret,
+        replicas: 1,
+        // TODO: check fastapi env variables
+        env: [],
```
Comment (judtinzhang) [inline k8s/main.ts:35]: `REDIS_URL`, `DATABASE_URL`, `POSTGRES_HOST`, `POSTGRES_PORT`

**Review by judtinzhang — COMMENTED** [review]
Looks great, thanks joy!

File: .github/build-deploy.yaml (hunk 2 above)
Comment (joyliu-q) [reply .github/build-deploy.yaml:5]: LMAO

Comment (joyliu-q) [reply .github/build-deploy.yaml:5]: Shhhh

**Pull request merged**